### PR TITLE
Added Implicit operators for Error and List<Errors>

### PR DESCRIPTION
**Pull request by RHaughton**

### CODE REVIEW

File: src/FluentResults/Results/Result.cs [hunk 1]
```diff
@@ -58,17 +57,16 @@ public Result<TNewValue> Bind<TNewValue>(Func<Result<TNewValue>> bind)
         {
             var result = new Result<TNewValue>();
             result.WithReasons(Reasons);
-            
-            if (IsSuccess)
-            {
+
+            if (IsSuccess) {
```
Comment (altmann) [inline src/FluentResults/Results/Result.cs:61]: Please change back all this styling changes.

File: src/FluentResults/Results/Result.cs [hunk 2]
```diff
@@ -176,15 +170,24 @@ public async ValueTask<Result> Bind(Func<ValueTask<Result>> action)
...
                 var converted = await action();
                 result.WithReasons(converted.Reasons);
             }
 
             return result;
         }
+
+        public static implicit operator Result(Error error)
```
Comment (altmann) [inline src/FluentResults/Results/Result.cs:182]: Please add some tests for these implicit conversations for Result and Result of T

File: src/FluentResults/Results/Result.cs [hunk 3]
```diff
@@ -176,15 +170,24 @@ public async ValueTask<Result> Bind(Func<ValueTask<Result>> action)
...
+        {
+            return Fail(error);
+        }
+
+        public static implicit operator Result(List<Error> errors)
```
Comment (altmann) [inline src/FluentResults/Results/Result.cs:187]: I hope there is no conflict with the public static implicit operator Result<TValue>(TValue value)
we will see in the tests.

**Comment by RHaughton** [comment]
Reverted the style editing (sorry for that).
I just added unit testing for implicit conversion of Error and list of Errors.
Everything seems to work!

**Comment by eltoncezar** [comment]
It would be nice if you documented this feature in the README also.

**Comment by RHaughton** [comment]
> It would be nice if you documented this feature in the README also.

done

**Comment by eltoncezar** [comment]
What you think about converting IReasons instead of just Errors?

**Comment by RHaughton** [comment]
Well with errors we know for sure it will be a fail results. Reasons can be errors or successes.

**Comment by altmann** [comment]
> What you think about converting IReasons instead of just Errors?

Is it possible to use ```IError``` instead of ```Error```? I think I can remember that implicit conversions and interfaces are not working in .net

Is it possible to use ```IEnumerable<Error or IError>``` instead of ```List<Error or IError>```?

**Comment by RHaughton** [comment]
No Unfortunately it doesn't work. I would also prefer the IEnumerable/IError option (I tried it at first).

You can use a List<IError> but the IError alone won't work.

This fails 
`public` static implicit operator Result(IError error) // doesn't work

This works but It is inconsistent with the first Error implicit Operator
`public` static implicit operator Result(List<IError> errors) // works but meh

**Comment by RHaughton** [comment]
we could maybe also add an implicit operator for an Error array (Error[]) ?

**Comment by altmann** [comment]
> we could maybe also add an implicit operator for an Error array (Error[]) ?

I think a list is okay. I will have a detail look in the next days and merge it.

**Comment by RHaughton** [comment]
What do you think of returning and Exception and implicitly convert it to and Fail Result with an exceptionalError?

**Comment by altmann** [comment]
Puh, good question. I think its a little bit too much magic. Maybe we can add it in the future if we miss it.

**Pull request merged**